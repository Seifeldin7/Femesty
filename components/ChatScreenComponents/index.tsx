export * from './ChatInputBox'
export * from './ChatMessage'