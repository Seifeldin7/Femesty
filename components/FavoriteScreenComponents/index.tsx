export * from "./FavoriteHzScroller";
export * from "./FavoriteLook";
