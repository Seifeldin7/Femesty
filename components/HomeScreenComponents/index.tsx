export * from "./FilterButton"
export * from "./HeaderTab"
export * from "./HorizontalScroller"
export * from "./LookMedia"
export * from "./SideBar"