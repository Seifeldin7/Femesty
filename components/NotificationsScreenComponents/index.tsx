export * from "./HiddenItem";
export * from "./VisibleItem";
