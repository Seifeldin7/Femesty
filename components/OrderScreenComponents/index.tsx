export * from './OrderItem'
export * from './OrderCard'
export * from './OrderList'