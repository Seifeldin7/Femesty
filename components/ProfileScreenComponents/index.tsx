export * from "./ProfileInformation"
export * from "./MenuList"