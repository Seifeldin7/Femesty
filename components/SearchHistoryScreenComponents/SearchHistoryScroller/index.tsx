import React from "react";
import { Text } from "react-native";
import { FlatList } from "react-native-gesture-handler";
import * as Animatable from "react-native-animatable";
import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
import { faHistory } from "@fortawesome/free-solid-svg-icons";
import { TouchableOpacity } from "react-native-gesture-handler";
import { FontStyles } from "../../../styles/Fonts";

import { styles } from "./styles";
import { componentProps } from "./types";

export const SearchHistoryScroller = ({ onpress, history }: componentProps) => {
  const renderItem = ({ item }: any) => {
    const onPress = () => {
      onpress(item);
    };
    return (
      <Animatable.View animation="fadeInUpBig" duration={1000}>
        <TouchableOpacity
          onPress={() => {
            onPress();
          }}
          style={styles.historyScroller}
        >
          <FontAwesomeIcon style={styles.icon} size={20} icon={faHistory} />
          <Text
            numberOfLines={1}
            ellipsizeMode={"clip"}
            style={[FontStyles.NormalText, { fontSize: 18 }]}
          >
            {item}
          </Text>
        </TouchableOpacity>
      </Animatable.View>
    );
  };
  return (
    <FlatList
      data={history}
      renderItem={renderItem}
      keyExtractor={(item) => history.indexOf(item).toString()}
      style={styles.container}
    />
  );
};