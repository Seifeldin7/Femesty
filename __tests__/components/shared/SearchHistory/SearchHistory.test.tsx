import React from "react";
import { render } from "react-native-testing-library";
import { SearchHistoryScroller } from "../../../../components/SearchHistoryScreenComponents";
import {View} from "react-native";

jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: ''
}))

jest.mock('react-native-gesture-handler', () => ({
    TouchableOpacity: '',
    FlatList: ''
}))


test('should render history', () => {
    let onpress = jest.fn();
    let history=['look','summer'];
    const tree = render(
        <View>
            <SearchHistoryScroller
            history={history} onpress={(item:any)=>{onpress}}
            />
        </View>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});