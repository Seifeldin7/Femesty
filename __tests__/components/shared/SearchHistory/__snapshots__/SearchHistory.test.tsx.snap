// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render history 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "lightgray",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <
      data={
        Array [
          "look",
          "summer",
        ]
      }
      keyExtractor={[Function]}
      renderItem={[Function]}
    />
  </View>
</View>
`;
